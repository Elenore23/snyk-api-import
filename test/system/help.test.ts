import { exec } from 'child_process';
import { sep } from 'path';
const main = './dist/index.js'.replace(/\//g, sep);

describe('`snyk-api-import help <...>`', () => {
  const OLD_ENV = process.env;
  process.env.SNYK_API = process.env.SNYK_API_TEST;
  process.env.SNYK_TOKEN = process.env.SNYK_TOKEN_TEST;

  afterAll(async () => {
    process.env = { ...OLD_ENV };
  });
  it('Shows help text as expected', (done) => {
    return exec(`node ${main} help`, (err, stdout) => {
      if (err) {
        throw err;
      }
      expect(err).toBeNull();
      expect(stdout.trim()).toMatchSnapshot();
      done();
    });
  });
});
