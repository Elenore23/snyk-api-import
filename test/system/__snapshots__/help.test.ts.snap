// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`\`snyk-api-import help <...>\` Shows help text as expected 1`] = `
"index.js

Kick off API powered import

Commands:
  index.js import         Kick off API powered import     [default] [aliases: i]
  index.js import:data    Generate data required for targets to be imported via
                          API to create Snyk projects.

  index.js list:imported  List all targets imported in Snyk for a given group &
                          source type. An analysis is performed on all current
                          organizations and their projects to generate this. The
                          generated file can be used to skip previously imported
                          targets when running the \`import\` command
  index.js orgs:create    Create the organizations in Snyk based on data file
                          generated with \`orgs:data\` command. Output generates
                          key data for created and existing organizations for
                          use to generate project import data.
  index.js orgs:data      Generate data required for Orgs to be created via API
                          by mirroring a given source.


Options:
  --version  Show version number                                       [boolean]
  --help     Show help                                                 [boolean]"
`;
